test/gas-analytics/SpokePool.Relay.ts

```diff
@@ -0,0 +1,71 @@
+import { toBNWei, SignerWithAddress, Contract, ethers, seedWallet } from "../utils";
+import { spokePoolFixture, getRelayHash, getFillRelayParams } from "../fixtures/SpokePool.Fixture";
+
+require("dotenv").config();
+
+let spokePool: Contract, weth: Contract, erc20: Contract;
+let depositor: SignerWithAddress;
+
+// Constants caller can tune to modify gas tests.
+const RELAY_COUNT = 10;
+const RELAY_AMOUNT = toBNWei("10");
+
+function constructRelayParams(relayTokenAddress: string, depositId: number) {
+  const { relayData } = getRelayHash(depositor.address, depositor.address, depositId, 1, relayTokenAddress);
+  return getFillRelayParams(relayData, RELAY_AMOUNT);
+}
+describe("Gas Analytics: SpokePool Relays", function () {
+  before(async function () {
+    if (!process.env.GAS_TEST_ENABLED) this.skip();
+  });
+
+  beforeEach(async function () {
+    [depositor] = await ethers.getSigners();
+    ({ spokePool, weth, erc20 } = await spokePoolFixture());
+
```

Comment:
Can we seed the recipients with tokens too? I think recipients will _usually_ not be an empty wallet, so I think a safer assumption is that they're pre-seeded. If we want to see both cases, we could make some of the cases below seed the recipient and some not.

Reply:
Sure